korrecte-lib/src/linters/lints/alb_ingress_instance.rs

```diff
@@ -0,0 +1,187 @@
+use crate::linters::{Group, KubeObjectType, Lint, LintSpec};
+
+use crate::kube::ObjectRepository;
+use crate::reporting::Finding;
+use crate::reporting::Reporter;
+use k8s_openapi::api::core::v1::{ServiceSpec, ServiceStatus};
+use k8s_openapi::api::networking::v1beta1::{IngressSpec, IngressStatus};
+use kube::api::{KubeObject, Object};
+use std::collections::HashSet;
+
+/// **What it does:** Checks that all ALB ingresses are linked to service which have compatible types
+/// with the ingress. When ingress is configures with target-type `instance`, only `NodePort` and `LoadBalancer`
+/// types are allowed; when is configured as `ip`, only `ClusterIP` services are allowed.
+///
+/// **Why is this bad?** ALB ingress controller will fail to create the associated ALB if services
+/// have incompatible types.
+///
+/// **Known problems:**
+///
+/// **References**
+/// - https://kubernetes-sigs.github.io/aws-alb-ingress-controller/guide/ingress/annotation/#target-type
+pub(crate) struct AlbIngressInstance<'a> {
+    object_repository: &'a dyn ObjectRepository,
+}
+
+impl<'a> AlbIngressInstance<'a> {
+    pub fn new(object_repository: &'a dyn ObjectRepository) -> Self {
+        AlbIngressInstance { object_repository }
+    }
+}
+
+impl<'a> Lint for AlbIngressInstance<'a> {
+    fn v1beta1_ingress(
+        &self,
+        ingress: &Object<IngressSpec, IngressStatus>,
+        reporter: &dyn Reporter,
+    ) {
+        let metadata = &ingress.meta().annotations;
+        let is_alb_ingress = metadata
+            .get("kubernetes.io/ingress.class")
+            .map(|class| class == "alb")
+            .unwrap_or(false);
```

Comment:
what about using map_or instead? (https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or)

Reply:
Thanks! I didn't know about this method.